src/core/lib/gprpp/match.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_CORE_LIB_GPRPP_MATCH_H
+#define GRPC_CORE_LIB_GPRPP_MATCH_H
+
+#include "absl/types/variant.h"
+#include "src/core/lib/gprpp/overload.h"
+
+namespace grpc_core {
+
+template <typename... Fs, typename... Ts>
+auto Match(absl::variant<Ts...>&& value, Fs... fs)
```

Comment:
Please add a comment showing how to use this.

Reply:
Done